src/ImageSharp/Image/Image.FromStream.cs

```diff
@@ -246,9 +241,19 @@ private static T WithSeekableStream<T>(Stream stream, Func<Stream, T> action)
 
             if (stream.CanSeek)
             {
+                if (config.ReadOrigin == ReadOrigin.Begin)
+                {
+                    stream.Position = 0;
+                }
+
                 return action(stream);
             }
 
+            if (config.ReadOrigin == ReadOrigin.Current)
```

Comment:
Shouldn't this be `config.ReadOrigin != ReadOrigin.Current` or `config.ReadOrigin == ReadOrigin.Begin`?

Reply:
If a stream is non-seekable then it's always gonna end up at 0 since we copy it to a `MemoryStream`. I thought best to throw rather than silently ignore.

Reply:
Maybe I don't understand what this method should do but shouldn't it copy the data from the stream to a memory stream to make it seekable? The ReadOrigin is an option for the input stream or also for the stream that we create?

Reply:
I feel we should be using this setting as more of a hint than a hard rule. Basically we use it as permission to reset streams when they are seekable and when they are not then we use our current logic of copy and read from there.